test/framework/entity.test.mjs

```diff
@@ -116,6 +117,51 @@ describe('Entity', function () {
 
     });
 
+    describe('#clone', function () {
+
+        it('clones an entity', function () {
+            const entity = new Entity('Test');
+            Object.keys(components).forEach((name) => {
+                entity.addComponent(name);
+            });
```

Comment:
Isn't this just an idiomatic for-in loop?

```suggestion
            for (const name in components) {
                entity.addComponent(name);
            }
```

Reply:
Haha, very good point @kungfooman. Thanks for pointing this out. I have updated the code based on your suggestion.